sklearn/tests/test_docstring_parameters.py

```diff
@@ -0,0 +1,312 @@
+"""
+Test mismatches between docstring and method/function signature.
+"""
+
+# Authors: Joel Nothman <joel.nothman@gmail.com>
+# License: BSD 3 clause
+
+
+# TODO inspect for Cython (see sagenb.misc.sageinspect)
+from __future__ import print_function
+
+import os
+import inspect
+import imp
+import sys
+import sklearn
+import warnings
+from pkgutil import walk_packages
+from nose import SkipTest
+try:
+    # FIXME This is to skip this import in Python 2.6
+    # To be removed when support for py2.6 ceases
+    from collections import Counter
+except ImportError:
+    pass
+from sklearn.utils.testing import assert_equal, assert_array_equal
+
+
+scikit_learn_path = os.getcwd().replace("/sklearn/tests", "")
+docscrape_path = 'doc/sphinxext/numpy_ext/docscrape.py'
+docscrape = imp.load_source('docscrape',
+                            os.path.join(scikit_learn_path, docscrape_path))
+
+
+def f1(non_default, default=1, default_2=None):
+    """A perfect function for testing
+
+    Parameters
+    ----------
+    non_default : A non default argument
+    default : A default argument, default val=1
+    default_2 : Another default argument, default val=None
+    """
+
+
+def f2(non_default, default=1, default_2=5):
+    """A test function with missing parameters in docstring
+
+    Parameters
+    ----------
+    default : type
+    """
+
+
+def f3(non_default, default=1, default_2=None):
+    """A test function with additional parameter in docstring.
+    
+    The default values in argspec initialized to None are not considered.
```

Comment:
Have modified the same!

> in which case this is not foolproof

Yes None defaults are excluded from checks... Also cython codes are not tested... Both of these could be dealt in another PR perhaps?